api/src/main/scala/uk/ac/wellcome/platform/api/responses/Responses.scala

```diff
@@ -14,7 +18,64 @@ case class ResultListResponse(
   pageSize: Int = 10,
   totalPages: Int = 10,
   totalResults: Int = 100,
-  results: Array[_ <: Any]
+  results: Array[_ <: Any],
+  prev: Option[String] = None,
+  next: Option[String] = None
 ) {
   @JsonProperty("type") val ontologyType: String = "ResultList"
 }
+
+object ResultListResponse {
+
+  private def getParamMapFromCustomCCRequestParams(cc: ApiRequest): ParamMap =
```

Comment:
There is some nasty reflection going on here to get back a Finagle ParamMap from our "Custom Request Case Class" - I couldn't find a publicly accessible way of doing this in the Finatra code.

This code will work for _single level_ case classes, but will fall over when asked to perform this operation on a nested case class.